### PR TITLE
Molecule class

### PR DESCRIPTION
**Pull request by kml5480**
Added a molecule class with psi4 geometries, basis set, number of electrons, and number of doubly occupied orbitals.

### CODE REVIEW

File: quest/molecule.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import psi4
```
Comment (dgasmith) [inline quest/molecule.py:5]: Can we add a header up top to describe what this file does?

File: quest/molecule.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    psi4 geometry, basis set, number of electrons, and number of doubly
+    occupied orbitals.
+    '''
+
+    def __init__(self, mol=None, bas=None):
+        '''
+        Initialize molecule with geometry mol and basis
+        set bas. Calculate number of electrons nel and
+        number of doubly occupied orbitals ndocc.
+        '''
```
Comment (dgasmith) [inline quest/molecule.py]: Can we add the NumPy style doc strings here?

File: quest/molecule.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        self.mol = mol
+        self.bas = bas
+
+        # Calculate the number of doubly occupied orbitals and the number of electrons
+        self.nel = sum(self.molecule.Z(n) for n in range(self.molecule.natom()))
+        self.nel -= self.molecule.molecular_charge()
+        
+        if not (self.nel / 2.0).is_integer():
+            raise ValueError("Molecule must have an even number of electrons to perform RHF.")
+        self.ndocc = int(self.nel / 2.0)
```
Comment (dgasmith) [inline quest/molecule.py:56]: You may need to call `set_geometry` and `set_basis` if these are not `None`.

`if mol is not None: set_geometry(mol)`.

**Review by dgasmith — COMMENTED** [review]
Looks good, don't forget to add this `__init__.py`.

**Comment by kml5480** [comment]
@dgasmith Fixed all comments

**Comment by dgasmith** [comment]
Awesome thanks!

**Pull request merged**